README.md

```diff
@@ -17,56 +17,98 @@ Simple HTTPS API backend service, which will get data in JSON format and send it
- - Start the server with the address to which emails need to be sent
+### Build
 
- `./feedback-api -from "no-reply@example.com" -to "foo@bar.org"`
+The application can be build as follows:
 
- or
+```
+go build
+```
+
+### Run
+
+The application directly compiled with Go can be started as follows:
+
+```
+./feedback-api \
+-from "no-reply@example.com" \
+-to foo@barbaz.com \
+-smtp-server smtp.abc.xyz \
+-smtp-port 465 \
+-http-port 8001
+```
+
+As a result, an HTTP server should be listenning on `localhost:8001` (CTRL+C to stop).
 
- `SMTP_USERNAME="foo@foobar.in" SMTP_PASSWORD="barbazquux" ./feedback-api -from "no-reply@example.com" -to foo@barbaz.com -smtp-server smtp.abc.xyz -smtp-port 465`
+### Configure
 
-If you are using `bash`, before typing in the command above, type a `SPC` character, so that the above command carrying the username and password won't get into the bash history.
-A server listens on `localhost:8001`.
+There are a few environment variables avaialble to configure the application:
 
- - Issue Post request:
+- SMTP_HELO: To announce a valid client hostname when the SMTP client contacts the server.
+- SMTP_USERNAME and SMTP_PASSWORD: To authenticate the SMTP client if the server supports it.
+- SMTP_USE_TLS: Set to `false` to disable TLS, which can be useful if you want to use a dummy SMTP server for local development.
+- SMTP_USE_INSECURE_TLS: Set to `true` to allow insecure TLS, which can be useful if you want to test with a self-sign certificate server in development env.
 
- `curl --request POST --header "Content-Type: application/json" --data '{"feedback":{"questions":[{"question":"q1","answer":"a1"},{"question":"q2","answer":"a2"}]}}' localhost:8001/v1/feedback`
+Example:
+
+```
+SMTP_USERNAME="foo@foobar.in" \
+SMTP_PASSWORD="barbazquux" \
+SMTP_USE_INSECURE_TLS=true \
+./feedback-api \
+-from "no-reply@example.com" \
+-to foo@barbaz.com \
+-smtp-server localhost \
+```
+
+IMPORTANT: If you are using `bash`, before typing in the command above, type a `SPC` character, so that the above command carrying the username and password won't get into the bash history.
 
-### Disable TLS
 
-You can disable TLS by setting the environment variable `SMTP_USE_TLS` to `"false"`. This can be useful if you want to use a dummy SMTP server for local development.
+### Test
 
-### Allow insecure TLS
+Once the application in running, a POST request can be sent:
 
-You can enable insecure TLS by setting the environment variable `SMTP_USE_INSECURE_TLS` to `"true"`. This can be useful if you want to test with a self-sign certificate server in development env.
+ ```
+ curl \
+--request POST \
+--header "Content-Type: application/json" \
+--data '{"feedback":{"title": "Full Feedback Form","rate": {"type" : "numbers","value" : 10},"questions":[{"question":"q1","answer":"a1"},{"question":"q2","answer":"a2"}]}}' \
+http://localhost:8001/v1/feedback
+```
+
+REM: The JSON content needs to matce the defined template!
```

Comment:
typo: `matce` -> `match` ?

Reply:
Thank you @vu3rdd.
BTW: where is the template, so I can add a link to it?

Reply:
https://github.com/LeastAuthority/feedback-api/blob/main/templates.go